src/components/DotNavigation/DotNavigation.tsx

```diff
@@ -0,0 +1,327 @@
+    prevState: DotNavigationState,
+  ) => {
+    const { currentIndex, length } = nextProps;
+    const { savedCurrentIndex } = prevState;
+
+    return DotNavigation.hasCurrentIndexChanged(
+      currentIndex,
+      savedCurrentIndex,
+      length,
+    )
+      ? {
+          ...prevState,
+          savedCurrentIndex: currentIndex,
+          ...(DotNavigation.shouldAnimate(
+            currentIndex,
+            savedCurrentIndex,
+            length,
+          ) && {
+            animation:
+              currentIndex < savedCurrentIndex
+                ? prevState.animation === Animation.Back
+                  ? Animation.BackNext
+                  : Animation.Back
+                : prevState.animation === Animation.Forward
+                ? Animation.ForwardNext
+                : Animation.Forward,
+          }),
+        }
+      : null;
+  };
+
+  static getDerivedStateFromProps = (
+    nextProps: DotNavigationProps,
+    prevState: DotNavigationState,
+  ) =>
+    nextProps.length <= MAX_SHORT_LIST_LENGTH
+      ? null
+      : DotNavigation.getLongListAnimation(nextProps, prevState);
+
+  getRadio = ({ checked, size, fakeRadio }: RadioProps) => (
+    <div
+      {...(fakeRadio && { tabIndex: -1, 'aria-hidden': true })}
+      className={classNames(styles.radio)}
+    >
+      <div
+        className={classNames(styles.dot, {
+          [styles.small]: size === Size.Small,
+          [styles.tiny]: size === Size.Tiny,
+          [styles.bordered]: this.props.showBorder === true,
+          [styles.light]: this.props.theme === Theme.Light,
+          [styles.checked]: checked,
+          [styles.noOpacityTransition]: this.props.noOpacityTransition,
+        })}
+        data-hook={fakeRadio ? undefined : DotNavigationDataHooks.Dot}
+      />
+    </div>
+  );
+
+  renderFakeRadio = () =>
+    this.getRadio({ checked: false, size: Size.Tiny, fakeRadio: true });
+
+  renderRadioButton = ({
+    key,
+    checked,
+    size,
+    onChange,
+    onKeyDown,
+    ariaLabel,
+  }: RadioButtonProps) => (
+    <RadioButton
+      key={key}
+      checked={checked}
+      checkedIcon={this.getRadio({ checked, size })}
+      uncheckedIcon={this.getRadio({ checked: false, size })}
+      onChange={onChange}
+      onKeyDown={onKeyDown}
+      aria-label={ariaLabel}
+      {...{ className: styles.extendedRadioButton }}
+    />
+  );
+
+  callOnSelect = (index: number) => () => this.props.onSelect(index);
+
+  startKeyDownHandler = (event: RadioButtonKeyDownEvent) =>
+    (event.key === Keys.ArrowLeft || event.key === Keys.ArrowUp) &&
+    event.preventDefault();
+
+  endKeyDownHandler = (event: RadioButtonKeyDownEvent) =>
+    (event.key === Keys.ArrowRight || event.key === Keys.ArrowDown) &&
+    event.preventDefault();
+
+  getKeyDownHandler = (value: number) =>
+    value === 0
+      ? this.startKeyDownHandler
+      : value === this.props.length - 1
+      ? this.endKeyDownHandler
+      : noop;
```

Comment:
It's better to set an `undefined` instead of `noop`.
`noop` will add an event listener, and `undefined` wont.

Reply:
Done) I received an error that undefined cannot be passed instead of a function before, re-installed node modules, fixed.